biz.aQute.repository/src/aQute/bnd/repository/fileset/FileSetRepository.java

```diff
@@ -87,11 +92,45 @@ private BridgeRepository getBridge() throws Exception {
 				return null;
 			}
 			ResourceBuilder rb = new ResourceBuilder();
-			try (JarFile jar = new JarFile(file)) {
+			try (Jar jar = new Jar(file)) {
 				Domain manifest = Domain.domain(jar.getManifest());
 				boolean hasIdentity = rb.addManifest(manifest);
 				if (!hasIdentity) {
-					return null;
+					String name;
+					String version;
+					Optional<POM> pom;
+
+					// Check if there's a module name
+					if (jar.getModuleName() != null) {
```

Comment:
I also think you should use the version from maven before falling back to any module version. That is, module version is the last hope.

Reply:
so do you mean to turn this around? maven before JPMS? I'm not sure how you want me to interleave the module info and the maven info here with this comment.

Reply:
To get the version of an artifact, use this order: Bundle-Version, maven version, module version.
All bundles have a bundle version, all maven artifacts have a maven version, not all modules have a module version. For module version, you may need to handle that the module version cannot be coerced into a bundle version (perhaps via a maven version).

Reply:
I updated the logic.